server/routes/api/subscriptions.ts

```diff
@@ -62,24 +67,25 @@ router.post("subscriptions.info", auth(), async (ctx) => {
     "Not a valid subscription event for documents"
   );
 
-  const subscriptions = await sequelize.transaction(async (transaction) => {
+  const subscription = await sequelize.transaction(async (transaction) => {
     const document = await Document.findByPk(documentId, { transaction });
```

Comment:
```suggestion
    const document = await Document.findByPk(documentId, { userId: user.id, transaction });
```

I'm surprised this works without `userId` being passed in here – it probably wouldn't on a private collection

Reply:
Ah I assumed passing in `userId` would return documents created by/touched by user, but looking into `Document` model, it seems `findByPk` has custom implementation that checks for permission rather than narrowing down search.

Reply:
It doesn't _check_ the permission – it makes sure that the related data is returned in the query so that `authorize` can work correctly against the user in all scenarios.

Reply:
Should the same apply here as well?
https://github.com/outline/outline/blob/ff4bb1ecbd8b075d7bf82bcd3f2b6997a0c51413/server/routes/api/subscriptions.ts#L106

Reply:
> It doesn't _check_ the permission – it makes sure that the related data is returned in the query so that `authorize` call can work correctly against the user in all scenarios.

Ah that makes sense

Now I understand what you meant by this
> surprised this works without `userId`

Thanks for the explanation